addons/escoria-core/ui_library/menus/load_save/load/load_game.gd

```diff
@@ -34,12 +34,13 @@ func refresh_savegames():
 		$VBoxContainer/ScrollContainer/slots.remove_child(slot)
 	
 	var saves_list = escoria.save_manager.get_saves_list()
-	for i in range(saves_list.keys().max() + 1):
-		if saves_list.has(i):
-			var save_data = saves_list[i]
-			var new_slot = slot_ui_scene.instance()
-			$VBoxContainer/ScrollContainer/slots.add_child(
-				new_slot
-			)
-			new_slot.set_slot_name_date(save_data["name"], save_data["date"])
-			new_slot.connect("pressed", self, "_on_slot_pressed", [i])
+	if not saves_list.empty():
+		for i in range(saves_list.keys().max() + 1):
```

Comment:
Replace this line with

```gdscript
for save_key in saves_list.keys():
```

then you can delete line 39. Replace "i" with "save_key" then.

Reply:
Does he not need `i` further down in the `.connect` call?

Reply:
> Does he not need `i` further down in the `.connect` call?

Yes. He can use `save_key` instead there.

Reply:
Ah, ok. So the keys are numeric, then?

Reply:
I believe so.

Reply:
@StraToN can you confirm? Just want to make sure as I do like @dploeger 's suggestion.

Reply:
Done